src/main/resources/jupyter/edit-mode.js

```diff
@@ -0,0 +1,396 @@
+const dialog = require('base/js/dialog')
+const utils = require("base/js/utils")
+
+define(() => {
+    // TEMPLATED CODE
+    // Leonardo has logic to find/replace templated values in the format $(...).
+    const googleProject = $(googleProject);
+    const clusterName = $(clusterName);
+
+    let modalOpen = false
+    let meta = {}
+        //this needs to be available so the loop can be cancelled where needed
+    let syncMaintainer = 1
+    let shouldExit = false
+
+    const syncIssueButtons = {
+        'Make a Copy': {
+            click: () => saveAs(),
+            'class': 'btn-primary'
+        },
+        'Reload the workspace version and discard your changes': {
+            click: () => updateLocalCopyWithRemote()
+        }
+    }
+
+    const lockIssueButtons = {
+        'Run in Playground Mode': {
+            click: () => openPlaygroundMode(),
+            'class': 'btn-primary'
+        },
+        'Make a Copy': {
+            click: () => saveAs()
+        }
+    }
+
+    const noRemoteFileButtons = {
+        'Continue working': {
+            click: () => {},
+            'class': 'btn-primary'
+        }
+    }
+
+    const modeBannerId = "notification_mode"
+    const lockConflictTitle = "File is in use"
+    const syncIssueTitle = "File versions out of sync"
+    const syncIssueBody = "Your version of this file does not match the version in the workspace. What would you like to do?"
+    const syncIssueNotFoundBody = "This file was either deleted or never was stored with us."
+
+    const leoUrl = '' //we are choosing to use a relative path here
+        // const leoUrl = 'http://localhost:8080' //for testing against local server
+    const welderUrl = leoUrl + `/proxy/${googleProject}/${clusterName}/welder`
+        // const welderUrl = leoUrl
+    const localizeUrl = welderUrl + '/objects'
+        // const checkMetaUrl = welderUrl + '/checkMeta'
+    const checkMetaUrl = welderUrl + '/objects/metadata'
+    const lockUrl = welderUrl + '/objects/lock'
+    const lastLockedTimer = 60000
+
+    const headers = {
+        'Content-Type': 'application/json',
+        'Access-Control-Allow-Headers': '*'
+    }
+
+    const basePayload = {
+        mode: "no-cors",
+        headers: headers
+    }
+
+    const jupyterBaseUrl = `/notebooks/${googleProject}/${clusterName}`
+    const jupyterContentsAPIUrl = jupyterBaseUrl + "/api/contents/"
+
+    function init() {
+        console.log('here in edit mode init')
+            // checkMeta()
```

Comment:
Be sure to revert this before merging - extension doesn't do anything without this